IncubatingIntegrationTest/SetTest.cs

```diff
@@ -257,6 +257,176 @@ public async Task SetAddBatchAsync_ElementsAreStringEnumerable_RefreshTtl()
+        Assert.Equal(CacheGetStatus.MISS, (await client.SetFetchAsync(cacheName, setName)).Status);
+    }
+
+    [Theory]
+    [InlineData(null, "my-set", "my-element")]
+    [InlineData("cache", null, "my-element")]
+    [InlineData("cache", "my-set", null)]
+    public async Task SetRemoveElementAsync_NullChecksString_ThrowsException(string cacheName, string setName, string element)
+    {
+        await Assert.ThrowsAsync<ArgumentNullException>(async () => await client.SetRemoveElementAsync(cacheName, setName, element));
+    }
+
+    [Fact]
+    public async Task SetRemoveElementAsync_ElementIsString_HappyPath()
+    {
+        var setName = Utils.NewGuidString();
+        var element = Utils.NewGuidString();
+
+        await client.SetAddAsync(cacheName, setName, element, false);
+
+        // Remove element that is not there -- no-op
+        await client.SetRemoveElementAsync(cacheName, setName, Utils.NewGuidString());
+        var fetchResponse = await client.SetFetchAsync(cacheName, setName);
+        Assert.Equal(CacheGetStatus.HIT, fetchResponse.Status);
+        var set = fetchResponse.StringSet();
+        Assert.Single(set);
+        Assert.Contains(element, set);
+
+        // Remove element
+        await client.SetRemoveElementAsync(cacheName, setName, element);
+        fetchResponse = await client.SetFetchAsync(cacheName, setName);
+        Assert.Equal(CacheGetStatus.MISS, fetchResponse.Status);
+    }
+
+    [Fact]
+    public async Task SetRemoveElementAsync_SetIsMissingElementIsString_Noop()
+    {
+        var setName = Utils.NewGuidString();
+        var element = Utils.NewGuidString();
+
+        // Pre-condition: set is missing
+        Assert.Equal(CacheGetStatus.MISS, (await client.SetFetchAsync(cacheName, setName)).Status);
+
+        // Remove element that is not there -- no-op
+        await client.SetRemoveElementAsync(cacheName, setName, Utils.NewGuidString());
+
+        // Post-condition: set is still missing
+        Assert.Equal(CacheGetStatus.MISS, (await client.SetFetchAsync(cacheName, setName)).Status);
+    }
+
+    [Fact]
+    public async Task SetRemoveElementsAsync_NullChecksElementsAreByteArray_ThrowsException()
+    {
+        var setName = Utils.NewGuidString();
+        var testData = new byte[][][] { new byte[][] { Utils.NewGuidByteArray(), Utils.NewGuidByteArray() }, new byte[][] { Utils.NewGuidByteArray(), null! } };
+
+        await Assert.ThrowsAsync<ArgumentNullException>(async () => await client.SetRemoveElementsAsync(null!, setName, testData[0]));
+        await Assert.ThrowsAsync<ArgumentNullException>(async () => await client.SetRemoveElementsAsync(cacheName, null!, testData[0]));
+        await Assert.ThrowsAsync<ArgumentNullException>(async () => await client.SetRemoveElementsAsync(cacheName, setName, (byte[][])null!));
+        await Assert.ThrowsAsync<ArgumentNullException>(async () => await client.SetRemoveElementsAsync(cacheName, setName, testData[1]));
```

Comment:
Is the reason why you couldn't do a similar table test here w/ `Theory`/`InlineData` annotation because test case required a more complex data setup w/ `testData`? Just curious.

Reply:
`InlineData` has limitations. You can use arrays, but not arrays of arrays, so no lists of `byte[]`. Because the `*Batch` or `RemoveElements` type methods accept a list of things, we have to back off to manual tests like this.

Reply:
Cool makes sense thanks!!